lib/internal/child_process.js

```diff
@@ -182,30 +180,30 @@ function ChildProcess() {
     //
     // - spawn failures are reported with exitCode < 0
     //
-    var syscall = self.spawnfile ? 'spawn ' + self.spawnfile : 'spawn';
+    var syscall = this.spawnfile ? 'spawn ' + this.spawnfile : 'spawn';
     var err = (exitCode < 0) ? errnoException(exitCode, syscall) : null;
```

Comment:
@Trott it looks that way. I want to investigate a little more because that code is a tangled mess. I should either be able to add a test or remove the code later today.